hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java

```diff
@@ -748,11 +748,24 @@ private StandardBasicTypes() {
 	 * The standard Hibernate type for mapping {@code float[]} to JDBC {@link org.hibernate.type.SqlTypes#VECTOR VECTOR},
 	 * specifically for embedding vectors like provided by the PostgreSQL extension pgvector.
 	 */
-	public static final BasicTypeReference<float[]> VECTOR = new BasicTypeReference<>(
-			"vector", float[].class, SqlTypes.VECTOR
+	public static final BasicTypeReference<double[]> VECTOR = new BasicTypeReference<>(
```

Comment:
Why was it changed from float[] to double[]? If the change is necessary shouldn't the comment be updated ? Can it be different types for PostgreSQL and Oracle ?

Reply:
You are right, in the JDBC docs, they said "The preferred array class for SQL to Java conversions for VECTOR is double[], as a double can store the information of any VECTOR value without losing information." https://docs.oracle.com/en/database/oracle/oracle-database/23/jajdb/oracle/jdbc/OracleType.html#VECTOR, but since PostgreSQL only supports float values, we can either create basicTypeReference with different types for each db, or just use float for both. what do you think ?

Reply:
Is there existing code that relies on Hibernate to convert a SQL VECTOR to a float[]? We don't want to break that code by changing it to a double[].

I'd recommend leaving float[] as the default mapping. I'm sure the Hibernate team has already thought about why float[] is a sensible default, and they chose it for a good reason.

I'm not an ML programmer, but FWIW: I think 32-bit precision represents a sensible default for ML programming.

For Oracle Database, we do support a VECTOR of 64-bit floating point numbers. However, Oracle JDBC can convert that data into 32-bit floats. It will apply a narrowing conversion, as specified in the "SQL to Java Conversions" section here: http://ojdbc.us.oracle.com/scratch/JDBC_Javadoc/MAIN/latest/oracle/jdbc/OracleType.html#VECTOR_FLOAT64

BTW, I'm the one who wrote the OracleType JavaDocs. While I did say that double[] is _preferred_, it does not mean it is _required_.

Reply:
We choose `float[]` as default because it's the datatype of choice for most ML models and pgvector also only supports float32 or smaller at this point. So yes, please revert this change.